integration-tests/environments/electron/jenkins.groovy

```diff
@@ -0,0 +1,69 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2019 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+def onLinux() {
+  return {
+    node('docker') {
+      // Unstash the files in the repository
+      unstash 'source'
+      docker.build(
+        'ci/realm-js:electron',
+        // Based on node:10, installing xvbf
+        '-f integration-tests/environments/electron/Dockerfile .'
+      ).inside(
+        '-e HOME=/tmp' // NPM will create folders in ~/.npm
+      ) {
+        // Unstash the package produced when packaging
+        dir('integration-tests') {
+          unstash 'package'
+        }
+        // Install the packaged version of realm into the app and run the tests
+        dir('integration-tests/environments/electron') {
+          sh 'npm install'
+          // Run both main and renderer tests catching any errors
+          def error = null;
+          // First the main process
+          try {
+            // Using xvfb to allow Electron to open a window
+            sh 'xvfb-run npm run test/main -- main-test-results.xml'
+          } catch (Exception e) {
+            error = e;
```

Comment:
This pattern seems to hide errors. E.g. if the renderer process fails, any failures from the main process will not be reported.

Reply:
I think it's okay. I want the build phase to fail if any of the two fail - but I want both of them to run even if the other fails. I cannot run in parallel, because that's not supported here. I am not that interested in the exception thrown, it most likely won't contain anything valuable. The details will be in the output in the console and the JUnit test report.